tests/integration/test_charm.py

```diff
@@ -127,13 +131,41 @@ async def test_invalid_flask_config(flask_app: Application, invalid_configs):
             assert invalid_config in unit.workload_status_message
 
 
+@pytest.mark.parametrize(
+    "update_config, excepted_config",
+    [
+        pytest.param({"flask_foo_str": "testing"}, {"FOO_STR": "testing"}, id="str"),
+        pytest.param({"flask_foo_int": 128}, {"FOO_INT": 128}, id="int"),
+        pytest.param({"flask_foo_bool": True}, {"FOO_BOOL": True}, id="bool"),
```

Comment:
As specified in a previous comment, we shouldn't remove the "flask_" prefix from the configuration, if a user specified "flask_foo_str" the envvar will be "FLASK_FLASK_FOO_STR"